File: Code/OfRectanglesAndSquares.Java/src/OfRectanglesAndSquares/RectangleWithRound.java
package OfRectanglesAndSquares;

/**
 * This class rounds Rectangle height and width on sets to the ceiling integer
 */
public class RectangleWithRound extends Rectangle {
    
    public RectangleWithRound(double height, double width) {
        super(height, width);
    }

    public void setHeight(double height) {
        super.setHeight(Math.ceil(height));
    }

    public void setWidth(double width) {
        super.setWidth(Math.ceil(width));
    }
}

File: Code/OfRectanglesAndSquares/OfRectanglesAndSquares/RectangleWithRound.cs
namespace OfRectanglesAndSquares
{
    /// <summary>
    /// This class rounds Rectangle height and width on sets to the ceiling integer
    /// </summary>
    public class RectangleWithRound : Rectangle
    {
        public RectangleWithRound(double height, double width) : base(height, width)
        {
        }

        public override void SetHeight(double height)
        {
            base.SetHeight(Math.Ceiling(height));
        }

        public override void SetWidth(double width)
        {
            base.SetWidth(Math.Ceiling(width));
        }
    }
}
